README.md

```diff
@@ -1,77 +1,151 @@
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.
+### Required variables
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+#### ad_integration_realm
 
-Example of setting the variables:
+Active Directory realm, or domain name to join
 
-```yaml
-template_foo: "oof"
-template_bar: "baz"
-```
+#### ad_integration_password
+
+The password of the user used to authenticate with when joining the machine to the realm.  Do not use cleartext - use Ansible Vault to encrypt the value.
+
+### Optional variables
+
+#### ad_integration_user
+
+The user name to be used to authenticate with when joining the machine to the realm.
+
+Default: Administrator
+
+#### ad_integration_join_to_dc
+
+an Active Directory domain controller's hostname (do not use IP address) may be specified to join via that domain controller directly.
+
+Default: Not set
+
+#### ad_integration_auto_id_mapping
+
+perform automatic UID/GID mapping for users and groups, set to `false` to rely on POSIX attributes already present in Active Directory.
+
+Default: true
+
+#### ad_integration_client_software
+
+Only join realms for which we can use the given client software. Possible values include sssd or winbind. Not all values are supported for all realms.
+
+Default: Automatic selection
 
-### Variables Exported by the Role
+#### ad_integration_membership_software
 
-This section is optional.  Some roles may export variables for playbooks to
-use later.  These are analogous to "return values" in Ansible modules.  For
-example, if a role performs some action that will require a system reboot, but
-the user wants to defer the reboot, the role might set a variable like
-`template_reboot_needed: true` that the playbook can use to reboot at a more
-convenient time.
+The software to use when joining to the realm. Possible values include samba or adcli. Not all values are supported for all realms.
 
-Example:
+Default: Automatic selection
 
-`template_reboot_needed` - default `false` - if `true`, this means
-a reboot is needed to apply the changes made by the role
+#### ad_integration_computer_ou
+
+The distinguished name of an organizational unit to create the computer account. It can be relative to the Root DSE, or a complete LDAP DN.
+
+Default: Default AD computer container
+
+#### ad_integration_manage_timesync
+
+If true, the ad_integration role will use fedora.linux_system_roles.timesync. Requires providing a value for `ad_integration_timesync_source` to use as a time source.
+
+Default: false
+
+#### ad_integration_timesync_source
+
+Hostname or IP address of time source to synchronize the system clock with. Only applicable if `ad_integration_manage_timesync` is true.
```

Comment:
Maybe the role should set `ad_integration_manage_timesync: true` if `ad_integration_timesync_source` is specified?

Reply:
That makes sense, fixed.